models/articles.models.js

```diff
@@ -2,7 +2,18 @@ const db = require("../db/connection");
 
 function fetchArticleById(article_id) {
   return db
-    .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .query(
+      `
+        SELECT a.* ,
+        ( SELECT count(*)::INT 
+          FROM comments c 
+          WHERE c.article_id = a.article_id 
+        ) AS comment_count
+        FROM articles a
+        WHERE a.article_id = $1;
+      `, 
+      [article_id]
```

Comment:
Could you achieve this with a JOIN between the two tables instead of a new query?